src/main/java/com/petfinder/controller/AdvertisementController.java

```diff
@@ -118,4 +80,49 @@ public String newAdvertisement(@RequestParam(required = false) String title,
          }
         return "addAdvertisement";
     }
+	
+	@ResponseBody
+	@RequestMapping(value = "/searchResult")
+	public ModelAndView getSearchResults( Model model,
+        @RequestParam(required = false) String adInfo,
+        @RequestParam(required = false) String petInfo,
+        @RequestParam(required = false) String locationInfo,
+        @RequestParam(required = false) String tagInfo,
+        @RequestParam(required = false) int page
+	) {
+        model.addAttribute("advertisements",
+                advertisementService.getSearchedAdvertisements(page-1, 20, adInfo, petInfo, locationInfo, tagInfo)
+        );
+        this.preparePagination(model, page);
+        model.addAttribute("adInfo", adInfo);
+        model.addAttribute("petInfo", petInfo);
+        model.addAttribute("locationInfo", locationInfo);
+        model.addAttribute("tagInfo", tagInfo);
+        return new ModelAndView( "searchResults" );
+    }
+	
+	private void preparePagination(Model model, int page)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method preparePagination() has an NPath complexity of 400](https://www.codacy.com/app/matikz1110/PetFinder/file/2642351523/issues/source?bid=3239464&fileBranchId=3321823#l104)